### PR TITLE
Bluetooth: controller: Handle ACAD in Periodic Advertising

### PR DESCRIPTION
**Pull request by cvinayak**
Handle ACAD field in when updating the Advertising Data in
Periodic Advertising PDU.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by ppryga-nordic — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 1]
```diff
@@ -728,12 +729,23 @@ static uint8_t adv_sync_hdr_set_clear(struct lll_adv_sync *lll_sync,
...
-
-	/* AdvData */
-	/* Calc previous tertiary PDU len */
-	ter_len_prev = ull_adv_aux_hdr_len_calc(ter_com_hdr_prev,
-						&ter_dptr_prev);
+	/* Calc previous ACAD len and update PDU len */
+	ter_len_prev = ter_dptr_prev - (uint8_t *)ter_com_hdr_prev;
+	if (ter_len_prev < (ter_com_hdr_prev->ext_hdr_len +
+			    offsetof(struct pdu_adv_com_ext_adv,
+				     ext_hdr_adv_data))) {
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:736]: Please put the right side of the comparison into a (new) auto variable that is also reused below to help readability. If I understand the code correctly, something like `ter_hdr_len_prev` would be a suitable name.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 2]
```diff
@@ -728,12 +729,23 @@ static uint8_t adv_sync_hdr_set_clear(struct lll_adv_sync *lll_sync,
...
+				offsetof(struct pdu_adv_com_ext_adv,
+					 ext_hdr_adv_data) - ter_len_prev;
+		ter_len_prev += acad_len_prev;
+		ter_dptr_prev += acad_len_prev;
+		ter_dptr += acad_len_prev;
+	} else {
+		acad_len_prev = 0;
+		ter_len_prev = offsetof(struct pdu_adv_com_ext_adv,
+					ext_hdr_adv_data);
+		ter_dptr_prev = (uint8_t *)ter_com_hdr_prev + ter_len_prev;
```
Comment (wopu-ot) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:747]: Why are `ter_len_prev` and `ter_dptr_prev` set here instead of keeping their values?

**Review by wopu-ot — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_sync.c]: done.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_sync.c]: Added comments in code.

**Review by carlescufi — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**